services/viva-ms/test/helpers/completions.test.js

```diff
@@ -0,0 +1,270 @@
+  test.each([
+    {
+      conditionOption: {
+        requested: [...requestedAllFalseList],
+        isRandomCheck: true,
+        isAttachmentPending: false,
+      },
+      expectedResult: getStatusByType(ACTIVE_RANDOM_CHECK_REQUIRED_VIVA),
+      description: `yet to be submitted, expect ${ACTIVE_RANDOM_CHECK_REQUIRED_VIVA}`,
+    },
+    {
+      conditionOption: {
+        requested: [...requestedAllFalseList],
+        isRandomCheck: true,
+        isAttachmentPending: true,
+      },
+      expectedResult: getStatusByType(ACTIVE_RANDOM_CHECK_SUBMITTED),
+      description: `submitted with attachments, expect ${ACTIVE_RANDOM_CHECK_SUBMITTED}`,
+    },
+    {
+      conditionOption: {
+        requested: [...requestedSomeTrueList],
+        isRandomCheck: true,
+        isAttachmentPending: false,
+      },
+      expectedResult: getStatusByType(ACTIVE_COMPLETION_REQUIRED_VIVA),
+      description: `one or more requested received, expect ${ACTIVE_COMPLETION_REQUIRED_VIVA}`,
+    },
+  ])('$description', ({ conditionOption, expectedResult }) => {
+    const results = getCompletionStatus(conditionOption);
+    expect(results).toEqual(expectedResult);
+  });
+});
+
+describe('Completions requested received (getCompletionStatus)', () => {
+  test.each([
+    {
+      conditionOption: {
+        requested: [...requestedSomeTrueList],
+        isRandomCheck: false,
+        isAttachmentPending: true,
+      },
+      expectedResult: getStatusByType(ACTIVE_COMPLETION_SUBMITTED),
+      description: `submitted with attachments, expect ${ACTIVE_COMPLETION_SUBMITTED}`,
+    },
+    {
+      conditionOption: {
+        requested: [...requestedSomeTrueList],
+        isRandomCheck: false,
+        isAttachmentPending: false,
+      },
+      expectedResult: getStatusByType(ACTIVE_COMPLETION_REQUIRED_VIVA),
+      description: `completion requested, expect ${ACTIVE_COMPLETION_REQUIRED_VIVA}`,
+    },
+  ])('$description', ({ conditionOption, expectedResult }) => {
+    const results = getCompletionStatus(conditionOption);
+    expect(results).toEqual(expectedResult);
+  });
+});
+
+describe('Completions random select (getCompletionState)', () => {
+  test.each([
+    {
+      conditionOption: {
+        requested: [...requestedAllFalseList],
+        isRandomCheck: true,
+        isAttachmentPending: false,
+      },
+      expectedResult: VIVA_RANDOM_CHECK_REQUIRED,
+      description: `set state to ${VIVA_RANDOM_CHECK_REQUIRED} the first time`,
+    },
+    {
+      conditionOption: {
+        requested: [...requestedAllFalseList],
+        isRandomCheck: true,
+        isAttachmentPending: true,
+      },
+      expectedResult: COMPLETIONS_PENDING,
+      description: `set state to ${COMPLETIONS_PENDING} when uploaded attachments`,
+    },
+  ])('$description', ({ conditionOption, expectedResult }) => {
+    const results = getCompletionState(conditionOption);
+    expect(results).toEqual(expectedResult);
```

Comment:
`expectedResult` is a string, so you should use `toBe` instead of `toEqual`. `toBe` is used for primitive values.